rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -134,6 +134,11 @@ Node::create_publisher(
       shared_publish_callback,
       intra_process_publisher_handle);
   }
+  if (rmw_trigger_guard_condition(notify_guard_condition_) != RMW_RET_OK) {
+    throw std::runtime_error(
+            std::string(
+              "Failed to notify waitset of change to node state: ") + rmw_get_error_string());
```

Comment:
Would it be helpful to customize the following error message to mention how the node state changed in order to distinguish them in case they happen?